plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -177,6 +177,34 @@ func PodMatchesNodeLabels(pod *api.Pod, node *api.Node) bool {
 	return selector.Matches(labels.Set(node.Labels))
 }
 
+// NoPodLabelConflict evaluates if a pod fits a node by:
+// 1) Comparing the pod's conflict labels against labels of all the other pods.
+// 2) Comparing all the other pods conflict labels with labels of this pod.
+func NoPodLabelConflict(pod *api.Pod, existingPods []*api.Pod, node string) (bool, error) {
+	for podIx := range existingPods {
+		if existingPods[podIx].Labels != nil {
+			for conflict, value := range pod.Spec.Conflicts {
+				if existingValue, ok := existingPods[podIx].Labels[conflict]; ok {
```

Comment:
@bgrant0607 
Thanks for the pointers and the feedback. I could switch it to using a label selector, but I have one concern with the label selector. With a map, its possible to specify if key1 == val1 OR key2 == val2, its a conflict. Label selector has an AND semantic, which wouldn't make this possible.

The OR semantic may be very useful for conflicts. The label selector operators that are likely to be used with conflicts are EqualsOperator, DoubleEqualsOperator, InOperator and ExistsOperator. Other than InOperator, the rest of them are covered by a map albeit with an OR semantic.

Let me know your thoughts on how to proceed.